### ISSUE TITLE
Failed to update registries due to package renaming/deletion

### ISSUE THREAD

**Issue by cuihaoleo** [sunoru/julia-mirror#1]
Hi, I am a maintainer of USTC mirrors. We are using your scirpt to maintain a Julia mirror.

Sadly our repo has been broken for long. I am trying to recover it. The log file indicates following error:
```
fatal: Not a git repository (or any of the parent directories): .git
fatal: Not a git repository (or any of the parent directories): .git
fatal: Not a git repository (or any of the parent directories): .git
[2019-01-24 00:33:51]ERROR:Failed to update registry: General
[2019-01-24 00:33:51]ERROR:Failed to update registries
Traceback (most recent call last):
  File "/julia-mirror/scripts/mirror_julia.py", line 585, in <module>
    main()
  File "/julia-mirror/scripts/mirror_julia.py", line 579, in main
    try_update('registries', update_registries, config, status)
  File "/julia-mirror/scripts/mirror_julia.py", line 565, in try_update
    raise e
  File "/julia-mirror/scripts/mirror_julia.py", line 560, in try_update
    update(config, status)
  File "/julia-mirror/scripts/mirror_julia.py", line 439, in update_registries
    raise e
  File "/julia-mirror/scripts/mirror_julia.py", line 433, in update_registries
    update_registry(config, status, name, config.registries[name])
  File "/julia-mirror/scripts/mirror_julia.py", line 412, in update_registry
    update_package_list(config, name, registry_dir)
  File "/julia-mirror/scripts/mirror_julia.py", line 385, in update_package_list
    packages[package_name][registry_name] = get_package_info(each_dir)
  File "/julia-mirror/scripts/mirror_julia.py", line 368, in get_package_info
    with open(os.path.join(package_dir, 'Package.toml')) as fi:
FileNotFoundError: [Errno 2] No such file or directory: '/data/registries/General/Q/QRUpdate/Package.toml'
```

I checked the git repo, and believed that the rename from "QRUpdate" to "QRupdate" by so-called "the great bikeshed rename + updates" (https://github.com/JuliaRegistries/General/commit/5e7610914e6193b9723ecbc13901e27001c470b4) caused this error.

Simply put, git wouldn't remove the dangling directory because julia-mirror had put a symlink in it, and julia-mirror thought there is a package but failed read its Package.toml, and so the error raised.

I think that any package deletion / renaming will lead to this error.

**Comment by sunoru** [comment]
It should be fixed by https://github.com/sunoru/julia-mirror/commit/b3c80a69eae42f005c3096bdf21b240e77ff56cb .

I noticed that you have manually delete the renamed/deleted folders, but there are still folders for the packages like https://mirrors.ustc.edu.cn/julia/packages/QRUpdate/General/ .

Use something like `python3 ./scripts/remove_packages.py /data/packages` to remove those folders.
